Error: node_modules/@types/babel__traverse/index.d.ts:1463:42 - error TS1005: '?' expected.

### Discussed in https://github.com/oppia/oppia/discussions/22279

<div type='discussions-op-text'>

<sup>Originally posted by **Harish-1511** March 27, 2025</sup>



```
harish05@harish05-VirtualBox:~/Documents/opensource/oppia$ python -m scripts.start

[notice] A new release of pip is available: 24.3.1 -> 25.0.1
[notice] To update, run: pip install --upgrade pip

[notice] A new release of pip is available: 24.3.1 -> 25.0.1
[notice] To update, run: pip install --upgrade pip

[notice] A new release of pip is available: 24.3.1 -> 25.0.1
[notice] To update, run: pip install --upgrade pip
Found existing installation: configparser 7.2.0
Uninstalling configparser-7.2.0:
  Successfully uninstalled configparser-7.2.0
Running install_third_party_libs script...
Node is installed.
Yarn is installed.
Redis-cli is already installed.
ElasticSearch is already installed.
Installing pre-commit hook for git

Symlink already exists
Making pre-commit hook file executable ...
pre-commit hook file is now executable!
Installing pre-push hook for git
Symlink already exists
Making pre-push hook file executable ...
pre-push hook file is now executable!
Installing third-party Python and JS libs in third_party directory

Checking if pip is installed on the local machine
Regenerating "requirements.txt" file...
The generated requirements file may be rejected by pip install. See # WARNING lines for details.
All third-party Python libraries are already installed correctly.
Google Cloud SDK is installed.
Installing third-party Node modules in node_modules directory

yarn install v1.22.15
warning ../../../package.json: No license field
[1/4] Resolving packages...
warning jest > @jest/core > jest-runner > jest-environment-jsdom > jsdom > w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
warning jest > @jest/core > jest-runner > jest-environment-jsdom > jsdom > domexception@2.0.1: Use your platform's native DOMException instead
warning jest > @jest/core > jest-runner > jest-environment-jsdom > jsdom > data-urls > abab@2.0.6: Use your platform's native atob() and btoa() methods instead
warning jest-image-snapshot > rimraf@2.7.1: Rimraf versions prior to v4 are no longer supported
[2/4] Fetching packages...
warning Pattern ["@definitelytyped/typescript-versions@latest"] is trying to unpack in the same destination "/home/harish05/Documents/opensource/yarn_cache/v6/npm-@definitelytyped-typescript-versions-0.0.140-f7dd6d67fba93b590f543ff65ce55e3d4fbbee04-integrity/node_modules/@definitelytyped/typescript-versions" as pattern ["@definitelytyped/typescript-versions@^0.0.140","@definitelytyped/typescript-versions@^0.0.140","@definitelytyped/typescript-versions@^0.0.140"]. This could result in non-deterministic behavior, skipping.
info fsevents@2.3.2: The platform "linux" is incompatible with this module.
info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@2.3.3: The platform "linux" is incompatible with this module.
info "fsevents@2.3.3" is an optional dependency and failed compatibility check. Excluding it from installation.
info @ffmpeg-installer/darwin-arm64@4.1.5: The platform "linux" is incompatible with this module.
info "@ffmpeg-installer/darwin-arm64@4.1.5" is an optional dependency and failed compatibility check. Excluding it from installation.
info @ffmpeg-installer/darwin-arm64@4.1.5: The CPU architecture "x64" is incompatible with this module.
info @ffmpeg-installer/darwin-x64@4.1.0: The platform "linux" is incompatible with this module.
info "@ffmpeg-installer/darwin-x64@4.1.0" is an optional dependency and failed compatibility check. Excluding it from installation.
info @ffmpeg-installer/linux-arm@4.1.3: The CPU architecture "x64" is incompatible with this module.
info "@ffmpeg-installer/linux-arm@4.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
info @ffmpeg-installer/linux-arm64@4.1.4: The CPU architecture "x64" is incompatible with this module.
info "@ffmpeg-installer/linux-arm64@4.1.4" is an optional dependency and failed compatibility check. Excluding it from installation.
info @ffmpeg-installer/linux-ia32@4.1.0: The CPU architecture "x64" is incompatible with this module.
info "@ffmpeg-installer/linux-ia32@4.1.0" is an optional dependency and failed compatibility check. Excluding it from installation.
info @ffmpeg-installer/win32-ia32@4.1.0: The platform "linux" is incompatible with this module.
info "@ffmpeg-installer/win32-ia32@4.1.0" is an optional dependency and failed compatibility check. Excluding it from installation.
info @ffmpeg-installer/win32-ia32@4.1.0: The CPU architecture "x64" is incompatible with this module.
info @ffmpeg-installer/win32-x64@4.1.0: The platform "linux" is incompatible with this module.
info "@ffmpeg-installer/win32-x64@4.1.0" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > @angular/elements@11.2.10" has incorrect peer dependency "@angular/core@11.2.10".
warning " > @angular/elements@11.2.10" has incorrect peer dependency "@angular/platform-browser@11.2.10".
warning " > @ng-bootstrap/ng-bootstrap@9.1.3" has unmet peer dependency "@angular/localize@^11.0.0".
warning " > ngx-joyride@2.3.1" has incorrect peer dependency "tslib@^1.10.0".
warning " > ngx-translate-cache@9.0.2" has incorrect peer dependency "@angular/common@^9.1.0".
warning " > ngx-translate-cache@9.0.2" has incorrect peer dependency "@angular/core@^9.1.0".
warning " > ngx-translate-cache@9.0.2" has incorrect peer dependency "tslib@^1.10.0".
warning "firebase-admin > @firebase/database-compat > @firebase/database > @firebase/auth-interop-types@0.1.7" has unmet peer dependency "@firebase/app-types@0.x".
warning " > ts-jest@28.0.8" has incorrect peer dependency "jest@^28.0.0".
warning " > ts-jest@28.0.8" has incorrect peer dependency "typescript@>=4.3".
warning Workspaces can only be enabled in private projects.
[4/4] Building fresh packages...
Done in 63.02s.
Building third party libs at third_party/generated/
Starting new Redis Server: /home/harish05/Documents/opensource/oppia/../oppia_tools/redis-cli-6.2.4/src/redis-server redis.conf
Starting new ElasticSearch Server: /home/harish05/Documents/opensource/oppia/../oppia_tools/elasticsearch-7.17.0/bin/elasticsearch -q
Starting new Firebase Emulator: /home/harish05/Documents/opensource/oppia/node_modules/firebase-tools/lib/bin/firebase.js emulators:start --only auth --project dev-project-id --config .firebase.json --export-on-exit /home/harish05/Documents/opensource/oppia/../firebase_emulator_cache
⚠  emulators: You are not currently authenticated so some features may not work correctly. Please run firebase login to authenticate the CLI.
i  emulators: Starting emulators: auth
i  ui: Emulator UI logging to ui-debug.log

┌─────────────────────────────────────────────────────────────┐
│ ✔  All emulators ready! It is now safe to connect your app. │
│ i  View Emulator UI at http://localhost:4000                │
└─────────────────────────────────────────────────────────────┘

┌────────────────┬────────────────┬────────────────────────────┐
│ Emulator       │ Host:Port      │ View in Emulator UI        │
├────────────────┼────────────────┼────────────────────────────┤
│ Authentication │ localhost:9099 │ http://localhost:4000/auth │
└────────────────┴────────────────┴────────────────────────────┘
  Emulator Hub running at localhost:4400
  Other reserved ports: 4500

Issues? Report them at https://github.com/firebase/firebase-tools/issues and attach the *-debug.log files.
 
Starting new Cloud Datastore Emulator: /home/harish05/Documents/opensource/oppia_tools/google-cloud-sdk-500.0.0/google-cloud-sdk/bin/gcloud beta emulators datastore start --project dev-project-id --data-dir /home/harish05/Documents/opensource/oppia/../cloud_datastore_emulator_cache --host-port localhost:8089 --consistency=1.0 --quiet --no-store-on-disk
WARNING: Cloud Datastore Emulator support for Java JRE version 11 will be dropped after gcloud command-line tool release 508.0.0. Please upgrade to Java JRE version 21 or higher to continue using the latest Cloud Datastore Emulator.
Executing: /home/harish05/Documents/opensource/oppia_tools/google-cloud-sdk-500.0.0/google-cloud-sdk/platform/cloud-datastore-emulator/cloud_datastore_emulator create --project_id=dev-project-id /home/harish05/Documents/opensource/oppia/../cloud_datastore_emulator_cache
[datastore] Mar 26, 2025 10:16:43 PM com.google.cloud.datastore.emulator.CloudDatastore$CreateAction$1 apply
[datastore] INFO: Provided project_id to Cloud Datastore emulator create command, which is no longer necessary.
[datastore] Created new Cloud Datastore project in '/home/harish05/Documents/opensource/oppia/../cloud_datastore_emulator_cache'.
Executing: /home/harish05/Documents/opensource/oppia_tools/google-cloud-sdk-500.0.0/google-cloud-sdk/platform/cloud-datastore-emulator/cloud_datastore_emulator start --host=localhost --port=8089 --store_on_disk=False --allow_remote_shutdown --consistency=1.0 /home/harish05/Documents/opensource/oppia/../cloud_datastore_emulator_cache
[datastore] Mar 26, 2025 10:16:44 PM com.google.cloud.datastore.emulator.CloudDatastore$FakeDatastoreAction$9 apply
[datastore] INFO: Provided --allow_remote_shutdown to start command which is no longer necessary.
[datastore] Mar 26, 2025 10:16:44 PM com.google.cloud.datastore.emulator.CloudDatastore$StartAction apply
[datastore] INFO: Running CLOUD_DATASTORE emulator.
[datastore] Mar 26, 2025 10:16:44 PM com.google.cloud.datastore.emulator.impl.LocalDatastoreFileStub <init>
[datastore] INFO: Local Datastore initialized:
[datastore] 	Type: High Replication
[datastore] 	Storage: In-memory
[datastore] API endpoint: http://localhost:8089
[datastore] If you are using a library that supports the DATASTORE_EMULATOR_HOST environment variable, run:
[datastore] 
[datastore]   export DATASTORE_EMULATOR_HOST=localhost:8089
[datastore] 
[datastore] Dev App Server is now running.
[datastore] 
[datastore] The previous line was printed for backwards compatibility only.
[datastore] If your tests rely on it to confirm emulator startup,
[datastore] please migrate to the emulator health check endpoint (/). Thank you!
[datastore] The health check endpoint for this emulator instance is http://localhost:8089/
Starting new Angular Compiler: /home/harish05/Documents/opensource/oppia/node_modules/.bin/ng build --watch

Compiling @angular/core : es2015 as esm2015

Compiling @angular/common : es2015 as esm2015

Compiling @angular/platform-browser : es2015 as esm2015

Compiling @angular/platform-browser-dynamic : es2015 as esm2015

Compiling @angular/common/http : es2015 as esm2015

Compiling @angular/animations : es2015 as esm2015

Compiling @angular/animations/browser : es2015 as esm2015

Compiling @angular/platform-browser/animations : es2015 as esm2015

Compiling ngx-cookie : es2015 as esm2015

Compiling ngx-toastr : es2015 as esm2015

Compiling @angular/router : es2015 as esm2015

Compiling @angular/forms : es2015 as esm2015

Compiling @ng-bootstrap/ng-bootstrap : es2015 as esm2015

Compiling @angular/upgrade/static : es2015 as esm2015

Compiling @ngx-translate/core : es2015 as esm2015

Compiling ngx-translate-cache : es2015 as esm2015

Compiling @ngx-translate/http-loader : es2015 as esm2015

Compiling @angular/elements : es2015 as esm2015

Compiling @angular/cdk/keycodes : es2015 as esm2015

Compiling @angular/cdk/platform : es2015 as esm2015

Compiling @angular/cdk/observers : es2015 as esm2015

Compiling @angular/cdk/a11y : es2015 as esm2015

Compiling @angular/cdk/bidi : es2015 as esm2015

Compiling @angular/material/core : es2015 as esm2015

Compiling @angular/material/button : es2015 as esm2015

Compiling @angular/youtube-player : es2015 as esm2015

Compiling @angular/cdk/portal : es2015 as esm2015

Compiling @ctrl/ngx-codemirror : es2015 as esm2015

Compiling @angular/cdk/collections : es2015 as esm2015

Compiling @angular/cdk/scrolling : es2015 as esm2015

Compiling @angular/cdk/drag-drop : es2015 as esm2015

Compiling @angular/fire : es2015 as esm2015

Compiling @angular/fire/auth : es2015 as esm2015

Compiling @angular/material/progress-spinner : es2015 as esm2015

Compiling ngx-joyride : es2015 as esm2015

Compiling @angular/cdk/overlay : es2015 as esm2015

Compiling @angular/material/menu : es2015 as esm2015

Compiling @angular/material/form-field : es2015 as esm2015

Compiling @angular/material/select : es2015 as esm2015

Compiling @angular/material/checkbox : es2015 as esm2015

Compiling @angular/material/icon : es2015 as esm2015

Compiling @angular/cdk/layout : es2015 as esm2015

Compiling @angular/material/tooltip : es2015 as esm2015

Compiling ngx-mat-select-search : es2015 as esm2015

Compiling @angular/material/card : es2015 as esm2015

Compiling @angular/material/chips : es2015 as esm2015

Compiling @angular/material/dialog : es2015 as esm2015

Compiling @angular/cdk/text-field : es2015 as esm2015

Compiling @angular/material/input : es2015 as esm2015

Compiling @angular/material/datepicker : es2015 as esm2015

Compiling @angular/cdk/accordion : es2015 as esm2015

Compiling @angular/material/expansion : es2015 as esm2015

Compiling @angular/material/grid-list : es2015 as esm2015

Compiling @angular/material/divider : es2015 as esm2015

Compiling @angular/material/list : es2015 as esm2015

Compiling @angular/material/progress-bar : es2015 as esm2015

Compiling @angular/material/radio : es2015 as esm2015

Compiling @angular/material/tabs : es2015 as esm2015

Compiling @angular/material/slider : es2015 as esm2015

Compiling @angular/material/slide-toggle : es2015 as esm2015

Compiling @angular/material/autocomplete : es2015 as esm2015

Compiling @asymmetrik/ngx-leaflet : module as esm5

Compiling ngx-trim-directive : es2015 as esm2015

Compiling @angular/cdk/clipboard : es2015 as esm2015

Compiling @angular/material/snack-bar : es2015 as esm2015

Compiling ng-circle-progress : es2015 as esm2015

Compiling @angular/material/paginator : es2015 as esm2015

Compiling @angular/material/button-toggle : es2015 as esm2015

Compiling @angular/material/sort : es2015 as esm2015

Compiling @angular/cdk/table : es2015 as esm2015

Compiling @angular/material/table : es2015 as esm2015

Compiling ngx-infinite-scroll : es2015 as esm2015
✔ Browser application bundle generation complete.

Warning: /home/harish05/Documents/opensource/oppia/core/templates/pages/lightweight-oppia-root/app.module.ts depends on 'hammerjs'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/core/templates/pages/common-imports.ts depends on 'core-js/es7/reflect'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/core/templates/pages/common-imports.ts depends on 'zone.js'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/core/templates/pages/exploration-editor-page/feedback-tab/services/thread-status-display.service.ts depends on 'lodash/cloneDeep'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/core/templates/domain/exploration/LostChangeObjectFactory.ts depends on 'lodash/isEqual'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/core/templates/services/audio-player.service.ts depends on 'howler'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/extensions/interactions/MathEquationInput/directives/math-equation-input-rules.service.ts depends on 'nerdamer'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/node_modules/mathjs/lib/esm/core/function/typed.js depends on 'typed-function'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/node_modules/mathjs/lib/esm/type/unit/Unit.js depends on '@babel/runtime/helpers/defineProperty'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/node_modules/mathjs/lib/esm/function/probability/util/seededRNG.js depends on 'seedrandom'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/node_modules/mathjs/lib/esm/function/arithmetic/floor.js depends on 'decimal.js'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/node_modules/mathjs/lib/esm/type/complex/Complex.js depends on 'complex.js'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/node_modules/mathjs/lib/esm/type/fraction/Fraction.js depends on 'fraction.js'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/core/templates/base-components/base.module.ts depends on 'components/common-layout-directives/common-elements/alert-message.component'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/core/templates/services/keyboard-shortcut.service.ts depends on 'mousetrap'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies



Error: node_modules/@types/babel__traverse/index.d.ts:1455:40 - error TS1005: '?' expected.

1455     K extends `${infer N extends number}` ? T[N]
                                            ~


Error: node_modules/@types/babel__traverse/index.d.ts:1463:42 - error TS1005: '?' expected.

1463     : K extends `${infer N extends number}`
                                              ~



✔ Browser application bundle generation complete.

Warning: /home/harish05/Documents/opensource/oppia/core/templates/components/shared-component.module.ts depends on 'pages/contributor-dashboard-page/modal-templates/translation-modal.component'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/node_modules/mathjs/lib/esm/type/bignumber/BigNumber.js depends on 'decimal.js'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Warning: /home/harish05/Documents/opensource/oppia/node_modules/mathjs/lib/esm/function/arithmetic/round.js depends on '@babel/runtime/helpers/defineProperty'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies



Error: node_modules/@types/babel__traverse/index.d.ts:1455:40 - error TS1005: '?' expected.

1455     K extends `${infer N extends number}` ? T[N]
                                            ~


Error: node_modules/@types/babel__traverse/index.d.ts:1463:42 - error TS1005: '?' expected.

1463     : K extends `${infer N extends number}`
                                              ~



^C 
i  emulators: Received SIGINT (Ctrl-C) for the first time. Starting a clean shutdown.
i  emulators: Please wait for a clean shutdown or send the SIGINT (Ctrl-C) signal again to stop right now.
i  Automatically exporting data using --export-on-exit "/home/harish05/Documents/opensource/oppia/../firebase_emulator_cache" please wait for the export to finish...
i  Found running emulator hub for project dev-project-id at http://localhost:4400
i  Exporting data to: /home/harish05/Documents/opensource/firebase_emulator_cache
i  emulators: Received export request. Exporting data to /home/harish05/Documents/opensource/firebase_emulator_cache.
✔  emulators: Export complete.
✔  Export complete
i  emulators: Shutting down emulators.
i  ui: Stopping Emulator UI
⚠  Emulator UI has exited upon receiving signal: SIGINT
i  auth: Stopping Authentication Emulator
i  hub: Stopping emulator hub
i  logging: Stopping Logging Emulator
Stopping Angular Compiler(name="sh", pid=4109)...


Servers are shutting down, please wait for them to end gracefully!


Stopping Cloud Datastore Emulator(name="sh", pid=4007)...
Terminated
Stopping Firebase Emulator(name="sh", pid=3947)...
Stopping ElasticSearch Server(name="sh", pid=3682)...
Stopping Redis Server(name="sh", pid=3674)...
Extending index.yaml...


Done! Thank you for waiting.


Traceback (most recent call last):
  File "/home/harish05/.pyenv/versions/3.9.20/lib/python3.9/contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/home/harish05/Documents/opensource/oppia/scripts/servers.py", line 360, in managed_redis_server
    yield proc
  File "/home/harish05/Documents/opensource/oppia/scripts/start.py", line 173, in main
    stack.enter_context(servers.managed_ng_build(watch_mode=True))
  File "/home/harish05/.pyenv/versions/3.9.20/lib/python3.9/contextlib.py", line 448, in enter_context
    result = _cm_type.__enter__(cm)
  File "/home/harish05/.pyenv/versions/3.9.20/lib/python3.9/contextlib.py", line 119, in __enter__
    return next(self.gen)
  File "/home/harish05/Documents/opensource/oppia/scripts/servers.py", line 435, in managed_ng_build
    for line in iter(read_line_func, None):
  File "/home/harish05/Documents/opensource/oppia/scripts/servers.py", line 432, in <lambda>
    lambda: proc.stdout.readline() or None
KeyboardInterrupt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/harish05/.pyenv/versions/3.9.20/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/harish05/.pyenv/versions/3.9.20/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/harish05/Documents/opensource/oppia/scripts/start.py", line 223, in <module>
    main()
  File "/home/harish05/Documents/opensource/oppia/scripts/start.py", line 219, in main
    dev_appserver.wait()
  File "/home/harish05/.pyenv/versions/3.9.20/lib/python3.9/contextlib.py", line 532, in __exit__
    raise exc_details[1]
  File "/home/harish05/.pyenv/versions/3.9.20/lib/python3.9/contextlib.py", line 517, in __exit__
    if cb(*exc_details):
  File "/home/harish05/.pyenv/versions/3.9.20/lib/python3.9/contextlib.py", line 405, in _exit_wrapper
    callback(*args, **kwds)
  File "/home/harish05/Documents/opensource/oppia/scripts/start.py", line 127, in call_extend_index_yaml
    extend_index_yaml.main()
  File "/home/harish05/Documents/opensource/oppia/scripts/extend_index_yaml.py", line 92, in main
    with open(WEB_INF_INDEX_XML_PATH, 'r', encoding='utf-8') as f:
FileNotFoundError: [Errno 2] No such file or directory: '/home/harish05/Documents/opensource/oppia/../cloud_datastore_emulator_cache/WEB-INF/appengine-generated/datastore-indexes-auto.xml'
```



Showing this error while executing the oppia anyone can please guide how should i overcome from this ?
</div>

--- a/package.json
+++ b/package.json
@@ -225,5 +225,11 @@
     "vanilla-picker": "^2.10.1",
     "wavesurfer.js": "^5.2.0",
     "wdio-intercept-service": "^4.2.2"
+  },
+  "resolutions": {
+    "@types/babel__traverse": "7.20.6"
+  },
+  "overrides": {
+    "@types/babel__traverse": "7.20.6"
   }
 }

--- a/package.json
+++ b/package.json
@@ -228,8 +228,5 @@
   },
   "resolutions": {
     "@types/babel__traverse": "7.20.6"
-  },
-  "overrides": {
-    "@types/babel__traverse": "7.20.6"
   }
 }

